drivers/ieee802154/ieee802154_rf2xx.c

```diff
@@ -230,26 +257,35 @@ static void rf2xx_thread_main(void *arg)
 		if (isr_status & (1 << RF2XX_RX_START)) {
 			ctx->trx_state = RF2XX_TRX_PHY_BUSY_RX;
 			k_timer_start(&ctx->trx_isr_timeout, K_MSEC(10), 0);
-		} else if (isr_status & (1 << RF2XX_TRX_END)) {
+
+			if (ctx->trx_model != RF2XX_TRX_MODEL_231) {
+				rf2xx_iface_sram_read(dev, 0, &ctx->rx_phr, 1);
+			}
+		}
+		if (isr_status & (1 << RF2XX_TRX_END)) {
 			if (ctx->trx_state == RF2XX_TRX_PHY_BUSY_RX) {
-				/* Ensures that automatically ACK will be sent
-				 * when requested
-				 */
-				while (rf2xx_iface_reg_read(dev,
-							    RF2XX_TRX_STATUS_REG) ==
-				       RF2XX_TRX_PHY_STATUS_BUSY_RX_AACK) {
-				};
-
-				/* Set PLL_ON to avoid transceiver receive
-				 * new data until finish reading process
-				 */
-				rf2xx_trx_set_state(dev,
-						    RF2XX_TRX_PHY_STATE_CMD_PLL_ON);
 				k_timer_stop(&ctx->trx_isr_timeout);
-				rf2xx_trx_rx(dev);
-				rf2xx_trx_set_state(dev,
-						    RF2XX_TRX_PHY_STATE_CMD_RX_AACK_ON);
-				/*if (ctx->trx_state == RF2XX_TRX_PHY_BUSY_TX)*/
+
+				if (ctx->trx_model != RF2XX_TRX_MODEL_231) {
+					rf2xx_trx_rx(dev);
+				} else {
+					/* Ensures that automatically ACK will be sent
+					* when requested
+					*/
+					while (rf2xx_iface_reg_read(dev,
```

Comment:
when you start to have too many if/for/while imbrication, it's time to refactor a bit for readability

Reply:
Done on fd54e41